src/datasources/db/v2/postgresql-logger.service.ts

```diff
@@ -0,0 +1,65 @@
+import { Injectable } from '@nestjs/common';
+import { AbstractLogger, type LogLevel, type LogMessage } from 'typeorm';
+
+@Injectable()
+export class PostgresqlLogger extends AbstractLogger {
+  /**
+   * Write log to specific output.
+   */
+  protected writeLog(
+    level: LogLevel,
+    logMessage: LogMessage | LogMessage[],
+  ): void {
+    const messages = this.prepareLogMessages(logMessage, {
+      highlightSql: false,
+    });
+
+    for (const message of messages) {
+      switch (message.type ?? level) {
+        case 'log':
+        case 'schema-build':
+        case 'migration':
+          console.log(message.message);
+          break;
+
+        case 'info':
+        case 'query':
+          if (message.prefix) {
+            console.info(message.prefix, message.message);
+          } else {
+            console.info(message.message);
+          }
+          break;
+
+        case 'warn':
+        case 'query-slow':
+          if (message.prefix) {
+            console.warn(message.prefix, message.message);
+          } else {
+            console.warn(message.message);
+          }
+          break;
+
+        case 'error':
+        case 'query-error':
+          if (message.prefix) {
+            console.error(message.prefix, message.message);
+          } else {
+            console.error(message.message);
+          }
```

Comment:
This is quite repetitive but I am also not a fan of the approach to simplify it. What do you think about abstracting it, e.g.

```ts
example(logger: (message?: unknown, ...rest: Array<unknown>) => void, message: LogMessage): void {
  if (message.prefix) {
    logger(message.prefix, message.message)
  } else {
    logger(message.message)
  }
}
```

Reply:
This is the implementation by TypeOrm, I prefer not to modify it since we only need to override the logQuery method.
https://orkhan.gitbook.io/typeorm/docs/logging

Reply:
Replied [above](https://github.com/safe-global/safe-client-gateway/pull/1988#discussion_r1789834553).